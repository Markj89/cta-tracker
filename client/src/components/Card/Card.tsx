/**
 * Card Component
 * Creates a dynamic div element
 * @type {Component} Card
 * @param CardProps
 * @returns JSX.Element
 */
import React, { useImperativeHandle, useRef } from "react";
import { CardProps } from "./Card.types";
import clsx from 'clsx';

const Card = React.forwardRef(({ children, style, className = '', station, onClick, orientation = 'vertical' }: CardProps, ref) => {
    const cardRef = useRef(null);
    
    useImperativeHandle(
        ref,
        () => ({
            focus: () => {
                cardRef?.current.focus();
            }
        }),
        []
    );
    return (
        <div ref={cardRef} className={clsx(orientation === 'vertical' ? 'card-vertical' : 'card-horizontal', className)}
         style={style} role="presentation">
            {children}
        </div>
    );
});

export default Card;