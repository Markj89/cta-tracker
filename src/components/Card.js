/**
 * Card 
 * @type {Component}
 */

import React from "react";
import styled from "styled-components";
import PropTypes from 'prop-types';


const StationCard = styled.div`
    position: relative;
    box-sizing: border-box;
    border-radius: 4px;
    background-color: white;
    box-sizing: -3px 0 5px rgba(36,41,46,.05);
`;
const TextWrapper = styled.div`
    padding-right: 15px;
    padding-left: 15px;

`;

const Card = ({ name, color, address }) => {
    return (
        <StationCard>
            <div>
                <div>
                    <div>
                        {{ name }}
                    </div>
                </div>
                <TextWrapper>
                    {{ address }}
                </TextWrapper>
            </div>
        </StationCard>
    );
}

Card.propTypes = {
    address: PropTypes.string,
    color: PropTypes.string,
    name: PropTypes.string,
};

export default Card;
