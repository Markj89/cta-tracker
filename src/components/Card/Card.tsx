/**
 * Card Component
 * 
 * @type {Component}
 */
import React, { forwardRef } from "react";
import { CardProps } from "./Card.types";

const Card = ({ children, className }: CardProps) => {
    return (
        <div className={className}>
            {children}
        </div>
    );
};