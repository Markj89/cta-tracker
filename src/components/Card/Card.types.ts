import { Station } from "components/Map";
import React, { Ref } from "react";

export interface CardProps {
    children?: React.ReactNode;
    className?: string;
    orientation?: string;
    station?: Station
    style?: React.CSSProperties;
    onClick?: () => void;
}