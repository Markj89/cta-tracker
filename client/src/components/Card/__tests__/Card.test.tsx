import { render } from '@testing-library/react';
import Card from '../Card';

describe('Card', () => {
    it('should render card component and match snapshot', () => {
        const wrapper = render(
            <Card>
                Hello World
            </Card>
        );
        expect(wrapper.container.firstChild).toMatchSnapshot();
    });
});