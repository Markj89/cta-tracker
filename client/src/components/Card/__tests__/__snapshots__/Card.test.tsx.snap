// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render card component and match snapshot 1`] = `
<div
  class="card-vertical"
>
  Hello World
</div>
`;
